File: Sample_Code/PHP/How_to_Use_a_Token.php
<?php
/**
* BluePay PHP 5 Sample Code
*
* This code sample runs a $3.00 Credit Card Auth transaction
* against a customer using test payment information.
* If approved, a 2nd transaction is run to capture the Auth.
* If using TEST mode, odd dollar amounts will return
* an approval and even dollar amounts will return a decline.
*/

include "BluePayPayment_BP10Emu.php";

$accountID = "MERCHANT'S ACCOUNT ID HERE";
$secretKey = "MERCHANT'S SECRET KEY HERE";
$mode = "TEST";

// Merchant's Account ID
// Merchant's Secret Key
// Transaction Mode: TEST (can also be LIVE)
$payment = new BluePayPayment_BP10Emu(
  $accountID,
  $secretKey,
  $mode);

// First Name: Bob
// Last Name: Tester
// Address1: 123 Test St.
// Address2: Apt #500
// City: Testville
// State: IL
// Zip: 54321
// Country: USA
$payment->setCustomerInformation(
  'Bob',
  'Tester',
  '123 Test St.',
  'Apt #500',
  'Testville',
  'IL',
  '61123',
  'USA');

// Card Number: 4111111111111111
// Card Expire: 12/15
// Card CVV2: 123
$payment->setCCInformation(
  '4111111111111111',
  '1215',
  '123');

// Phone #: 123-123-1234
$payment->setPhone('1231231234');

// Email Address: test@bluepay.com
$payment->setEmail('test@bluepay.com');

/* RUN A $3.00 CREDIT CARD AUTH */
$payment->auth('3.00');

$payment->process();

// If transaction was approved..
if ($payment->getStatus() == "APPROVED") {

  $paymentCapture = new BluePayPayment_BP10Emu(
    $accountID,
    $secretKey,
    $mode);

  // Attempts to capture above Auth transaction
  $paymentCapture->capture($payment->getTransID());

  $paymentCapture->process();

  // Read response from BluePay
  echo 'Status: '. $payment->getStatus() . '<br />' .
  'Message: '. $payment->getMessage() . '<br />' .
  'Transaction ID: '. $payment->getTransID() . '<br />' .
  'AVS Response: ' . $payment->getAVSResponse() . '<br />' .
  'CVS Response: ' . $payment->getCVV2Response() . '<br />' .
  'Masked Account: ' . $payment->getMaskedAccount() . '<br />' .
  'Card Type: ' . $payment->getCardType() . '<br />' .
  'Authorization Code: ' . $payment->getAuthCode() . '<br />';
} else {
  echo $payment->getMessage();
}
?>

File: Sample_Code/Perl/How_to_Use_a_Token.pl
##
# BluePay Perl Sample code.
#
# This code sample runs a $3.00 Credit Card Auth transaction
# against a customer using test payment information.
# If approved, a 2nd transaction is run to capture the Auth.
# If using TEST mode, odd dollar amounts will return
# an approval and even dollar amounts will return a decline.
##

use BluePay::BluePayPayment_BP10Emu;
use strict;

my $ACCOUNT_ID = "MERCHANT'S ACCOUNT ID HERE";
my $SECRET_KEY = "MERCHANT'S SECRET KEY HERE";
my $MODE = "TEST";

my $payment = BluePay::BluePayPayment_BP10Emu->new();

# Merchant values
$payment->{MERCHANT} = $ACCOUNT_ID;
$payment->{SECRET_KEY} = $SECRET_KEY;
$payment->{MODE} = $MODE;

# Customer values
$payment->{NAME1} = 'Bob';
$payment->{NAME2} = 'Tester';
$payment->{ADDR1} = '123 Test St.';
$payment->{ADDR2} = 'Apt #500';
$payment->{CITY} = 'Testville';
$payment->{STATE} = 'IL';
$payment->{ZIPCODE} = '54321';
$payment->{COUNTRY} = 'USA';

# Payment values
$payment->{TRANSACTION_TYPE} = 'AUTH';
$payment->{PAYMENT_TYPE} = 'CREDIT';
$payment->{AMOUNT} = '3.00';
$payment->{CC_NUM} = '4111111111111111';
$payment->{CC_EXPIRES} = '0815';
        
$payment->Post();

# If transaction was approved..
if($payment->{Result} == 'APPROVED') {

my $payment_capture = BluePay::BluePayPayment_BP10Emu->new();

$payment_capture->{MERCHANT} = $ACCOUNT_ID;
        $payment_capture->{SECRET_KEY} = $SECRET_KEY;
$payment_capture->{MODE} = $MODE;

# Attempts to capture transaction above
        $payment_capture->{TRANS_TYPE} = 'CAPTURE';

        $payment_token->Post();

# Get response from BluePay
print "TRANSACTION ID: " . $payment->{RRNO} . "\n";
print "STATUS: " . $payment->{Result} . "\n";
print "TRANSACTION MESSAGE: " . $payment->{MESSAGE} . "\n";
print "AVS RESULT: " .$payment->{AVS} . "\n";
print "CVV2 RESULT: " . $payment->{CVV2} . "\n";
print "MASKED PAYMENT ACCOUNT: " . $payment->{PAYMENT_ACCOUNT} . "\n";
print "CARD TYPE: " . $payment->{CARD_TYPE} . "\n";
print "AUTH CODE: " . $payment->{AUTH_CODE} . "\n";
} else {
print $payment->{MESSAGE};
}